azure-functions/acbs-function/src/functions/activity-update-facility-covenant.js

```diff
@@ -15,33 +15,23 @@
  * ------------------
  */
 const df = require('durable-functions');
-const { getNowAsIsoString } = require('../../helpers/date');
 const api = require('../../api');
+const { getNowAsIsoString } = require('../../helpers/date');
 const { isHttpErrorStatus } = require('../../helpers/http');
 
-/**
- * This function is used to update the covenant of a facility. It first checks if the payload is valid and contains all mandatory fields.
- * If the payload is valid, it sends a request to the API to update the facility covenant.
- * If the API request is successful, it returns an object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
- * If the API request fails, it throws an error with details about the request and the error.
- * If the payload is not valid or does not contain all mandatory fields, it returns an object with the missing mandatory fields.
- * If any other error occurs, it logs the error and throws a new error.
- *
- * @param {object} payload - The payload containing the facilityIdentifier and acbsFacilityCovenantInput.
- * @param {string} payload.facilityIdentifier - The identifier of the facility.
- * @param {import('../../mappings/facility/facility-covenant-amend').MappedFacilityCovenantAmendment} payload.acbsFacilityCovenantInput - The acbsFacilityCovenantInput object containing the covenant details.
- * @returns {object} - An object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
- * @throws {Error} - Throws an error if the payload is invalid, if the API request to generate the covenant ID fails, or if any other error occurs.
```

Comment:
are we worriedabout the loss of typing?

Reply:
I mean that the use of `imports` allow for reusable types -- so `acbsFacilityCovenantInput` is fully typed with intellisense etc and allows us to reuse this throughout (meaning we don't have to repeat definitions everywhere they're used)